nova_plugin/keypair.py

```diff
@@ -163,6 +163,15 @@ def validate_path_owner(path):
                 private_key_path)
             ctx.logger.error('VALIDATION ERROR: ' + err)
             raise NonRecoverableError(err)
+        else:
+            err = 'private key directory {0} is not writable'
```

Comment:
If we have existing path in OS '/a/b' and private key is provided as '/a/b/c/d/e' then, so what I understand, that what I need to do is to check if 'b' dir is writable, 'c' and 'd' don't exists. Your solution will check only 'd' dir (which doesn't exist so os.access will always return false) in this scenario. In my solution I iterate key path backwards looking for first dir that exists and only it is checked.